includes/class-attachments.php

```diff
@@ -446,7 +446,7 @@ private static function save_attachment( $attachment_data, $post_id, $author_id
 
 		// Prepare file array for WordPress.
 		$file_array = array(
-			'name'     => \basename( $attachment_data['url'] ),
+			'name'     => \basename( \strtok( $attachment_data['url'], '?' ) ),
```

Comment:
I am not sure how likely this is, but should we also strip anchors (`example.com#abc`)?

Reply:
Fixed in [7ff2eb1](https://github.com/Automattic/wordpress-activitypub/pull/2499/commits/7ff2eb1666f81f7e2d33450cccc66e28cad6a461).